doc/src/specifications/platforms/README.md

```diff
@@ -1,19 +1,51 @@
+
+Therefore, in the client-side portion of a psibase app, the cross-platform plugin is separated from the platform-native user interface. This separation allows for functionality to be reused between platforms, making the user-interfaces smaller/lighter than they would be otherwise.
+
+# Platform-specific app requirements
+
+The only part of a psibase app that is not cross-platform is the user interface. If a package published on a psibase network is intended to interface directly with a user, then it must deploy a user-interface that meets the specific requirements of the platform(s) on which it should be usable.
+
+## Entrypoint
 
-# Platform requirements
+The user-interface must have a platform-specific entrypoint. The http server for the app must serve an asset at a pre-defined path that is the accessed from the platform.
 
-For psibase apps to be functional on a particular platform, the app must publish a user-interface that works natively on the platform. Additionally, the network must serve a component that serves as the client-side wasm host. Among other things, this host is responsible for downloading all of the psibase app plugins (webassembly components) and preparing them for native execution. For example, since plugins are WebAssembly components, and components do not run natively (yet) in the web browser, the psibase web platform host must download the components, transpile them into core wasm and js, bundle them, and dynamically load them into the browser before plugin functionality can be executed in the browser.
+## User prompts
 
-Many changes and improvements to the platform host can be made without disrupting to the functioning of the psibase apps.
+User prompts are platform-specific elements of a user-interface that may be executed by a plugin. Therefore, though the plugin itself can run across platforms, the execution may fail at runtime if the platform host is unable to satisfy a requested user prompt. User prompts, like the entrypoint, are also typically handled by the app's http server serving an asset to facilitate the user-interaction at a predefined path.
 
-# Security implications
+# Platform-native host
+
+The platform must provide a wasm host that runs natively on the platform (client-side). Among other things, this host is responsible for downloading all of the psibase app plugins (WebAssembly components) and preparing them for native execution. For example, since plugins are WebAssembly components, and components do not run natively (yet) in the web browser, the psibase web platform host must download the components, transpile them into core wasm and js, bundle them, and dynamically load them into the browser before plugin functionality can be executed in the browser.
```

Comment:
This is getting a bit into the weeds. I would just say that the platform is responsible for downloading and executing the plugins.